docs/guides/using-the-network-shim.md

```diff
@@ -0,0 +1,288 @@
+        -   **[Support file](#support-file)**
+    -   **[Package scripts and dependencies](#package-scripts-and-dependencies)**
+    -   **[GitHub workflow](#github-workflow)**
+
+## About
+
+The Network Shim makes it possible to record and play back network traffic during a cypress tests run. This means no live network requests need to be made when running e2e tests on CI, which is generally considered an anti-pattern as it can lead to flaky tests.
+
+It consists of a cypress plugin as well as a setup utility to be placed in cypress' support file. These have to be used in conjunction. Under the hood it uses the cypress `cy.intercept()` API to intercept network traffic during a test run. It has three different _modes_:
+
+-   `capture`: when operating in this mode, the Network Shim will intercept all outgoing requests to the specified hosts and allow each request to reach the destination server. Each response is stored as a fixture (JSON file).
+-   `stub`: in this mode the Network Shim also intercepts all outgoing requests to the specified hosts. However, it will prevent the request from reaching the destination server and instead it will reply with a response from a previously recorded fixture.
+-   `live`: the Network Shim does not intercept any requests when this mode is enabled.
+
+By first recording fixtures during a _capture run_ and then using these fixtures during a _stub run_ it effectively becomes possible to execute a full cypress test suite without a live backend. This is especially relevant for running a cypress test suite on CI.
+
+By recording fixtures during a _capture run_ and then using these fixtures during a _stub run_ a full cypress test suite can be executed without a live backend. This is especially useful when running cypress tests on CI. Since it is no longer necessary to have a network connection and a backend, the cypress tests on CI can be more stable. One caveat is that the tests are slightly less realistic, since the network traffic is mocked, so the developer will have to make sure that the recorded fixtures are kept up-to-date.
+
+A capture or stub run is executed for a specific DHIS2 api version and individual fixtures for each version are recorded/played-back during a run. This makes it easier to do cross-version testing for _feature-toggling_ apps.
+
+## Quick setup
+
+The command-line-tool in the package `@dhis2/cli-utils-cypress` comes with an `install` command which generates most of the configuration files documented below. See the [Setup Cypress for your project](https://cli-utils-cypress.dhis2.nu/#/guides/setting-up-cli-tool) guide for more info on how to do this.
+
+To get up-and-running quickly do the following:
+
+1.  Install the cli tool
+
+    ```bash
+    yarn add --dev @dhis2/cli-utils-cypress
+    ```
+
+    <br/>
+
+1.  Issue the install command to generate a working setup
+
+    ```bash
+    d2-utils-cypress install all
+    ```
+
+    (_Instead of `all` another "group" can be specified._)
```

Comment:
Maybe link to where users can find out what those groups are, and what they (the groups) mean.

Reply:
I don't think that's needed, that link is already in the section intro.

Reply:
Ok, yeah I find that nice personally. Just a `see here` link makes it very clear where you can find the info when skimming.